caproto/_data.py

```diff
@@ -621,33 +699,39 @@ def _is_eligible(self, ss, flags, pair):
         # instance state so that self.subscribe can also use it.
         self._content.clear()
 
-        for queue, data_types in self._queues.items():
+        for queue, syncs in self._queues.items():
             # queue belongs to a Context that is expecting to receive
             # updates of the form (sub_specs, metadata, values).
             # data_types is a dict grouping the sub_specs for this queue by
             # their data_type.
-            for data_type, sub_specs in data_types.items():
-                eligible = tuple(ss for ss in sub_specs
-                                 if self._is_eligible(ss, flags, pair))
-                if not eligible:
-                    continue
-                try:
-                    metdata, values = self._content[data_type]
-                except KeyError:
-                    # Do the expensive data type conversion and cache it in
-                    # case another queue or a future subscription wants the
-                    # same data type.
-                    metadata, values = await self._read(data_type)
-                    self._content[data_type] = metadata, values
-
-                # We have applied the deadband filter on this side of the
-                # queue, deciding while SubscriptionSpecs should get this
-                # update. We will apply the array filter on the other side of
-                # the queue, since each eligible SubscriptionSpec may want a
-                # different slice. Sending the whole array through the queue
-                # isn't any more expensive that sending a slice; this is just a
-                # reference.
-                await queue.put((eligible, metadata, values))
+            for sync, data_types in syncs.items():
+                for data_type, sub_specs in data_types.items():
+                    eligible = tuple(ss for ss in sub_specs
+                                     if self._is_eligible(ss))
+                    if not eligible:
+                        continue
+                    if sync is None:
+                        channel_data = self
+                    else:
+                        try:
+                            channel_data = self._snapshots[sync.s][sync.m]
```

Comment:
Should we consider renaming the `s` and `m` to have some meaning, despite the filter-JSON names?